storage/badger/computation_result.go

```diff
@@ -0,0 +1,59 @@
+package badger
+
+import (
+	"errors"
+
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/badger/operation"
+)
+
+type ComputationResultUploadStatus struct {
+	db *badger.DB
+}
+
+func NewComputationResultUploadStatus(db *badger.DB) *ComputationResultUploadStatus {
+	return &ComputationResultUploadStatus{
+		db: db,
+	}
+}
+
+func (c *ComputationResultUploadStatus) Upsert(executionDataId flow.Identifier,
+	wasUploadCompleted bool) error {
+	_, err := c.ByID(executionDataId)
+	if errors.Is(err, storage.ErrNotFound) {
```

Comment:
If we insist on having upset, we should create a function for this, along `insert` and `update`, since querying for value is not performant, and badger really works as upset by default.

Reply:
done at PR #3062